WordPress/gutenberg
# Rename 'page content lock' to 'page content focus'

## What?
Renames all the new APIs added in https://github.com/WordPress/gutenberg/pull/50857 from 'page content lock' to 'page content focus'.

## Why?
I don't want to use the word "lock" since this feature no longer has anything to do with `templateLock` (it used to).

I did want to move away from "page" towards "post" as that matches the general convention in WordPress that everything is a "post", but, on second thought, I don't think we should do this because `@wordpress/edit-site` already has the stable `setPage` selector which I think we should align with.

## How?
Find and replace. So long as this makes it into Gutenberg 16.0 we don't need to do deprecations.

## Testing Instructions
Same as in https://github.com/WordPress/gutenberg/pull/50857.

1. Go to _Appearance → Editor → Pages_ and select a page.
1. Test the new switching functionality.
1. Check that navigating to _Appearance → Editor → Templates_ and selecting a template results in the same site editor experience as before (no switching functionality).
